### ISSUE TITLE
Regression coefficient for LGBMRegressor

### ISSUE THREAD

**Issue by Lucienxhh** [microsoft/LightGBM#5558]
## Summary
I use `lightgbm.LGBMRegressor` for regression prediction. however, i cannot find any API for regression coefficient.

## Motivation
Provide regression coefficient for `lightgbm.LGBMRegressor`, as the name **LGBMRegressor** implies.

**Comment by jameslamb** [comment]
Thanks for using LightGBM!

LightGBM does not fit a regression model like `y = mx + b` the way you may be used to with something like linear regression. Instead, it fits a sequence of trees.

So there is not a concept of "regression coefficients" for LightGBM the way that you might be used to with other types of regression.

If you're new to gradient boosting with trees, XGBoost's docs have an excellent guide: https://xgboost.readthedocs.io/en/stable/tutorials/model.html.

For LightGBM-specific details, see https://lightgbm.readthedocs.io/en/v3.3.2/Features.html.
